src/main/kotlin/com/autonomousapps/subplugin/ProjectPlugin.kt

```diff
@@ -835,6 +839,13 @@ internal class ProjectPlugin(private val project: Project) {
     )
   }
 
+  /** Get the buildPath of the current build from the root component of the resolution result. */
+  private fun Project.buildPath(configuration: String) = project.provider {
+    // Note: starting with Gradle 7.4, we can replace 'project.provider' with 'resolutionResult.rootComponent.map'
```

Comment:
what's this note about? does that relate to Gradle's CC support for accessing the dependency graph?

Reply:
The new Provider bases API – resolutionResult.rootComponent (as alternative to resolutionResult.root) – was added in that context I believe. But  doing it like we do now also works is you want to support older Gradle versions.

I use the current solution in another plugin as well and it is CC compatible.
[ExtraJavaModuleInfoPlugin.java#L131-L134](https://github.com/gradlex-org/extra-java-module-info/blob/main/src/main/java/org/gradlex/javamodule/moduleinfo/ExtraJavaModuleInfoPlugin.java#L131-L134)

Something like that is probably also what you would need to do here in the places where you now pass "Configurations" into tasks.